### ISSUE TITLE
[Experiment][Product][Architecture]: Remove the legacy generic research-workflow service graph from the default math toolbox

### ISSUE THREAD

**Issue by morluto** [morluto/jacobian#1226]
## Summary

Jacobian's current product definition says it is a toolbox of atomic mathematical operations, that agents own multi-step research strategy, and that the runtime is **not** a workflow engine. The default runtime still constructs and installs an older generic research-workflow platform beneath that product:

```text
claims
plugins
candidate evaluation
witness search
semantic enumeration
long-lived experiments
search orchestration
conjecture ingestion
transformation/preservation
shrinking
reference-domain manifests
```

Every normal runtime builds this complete `ApplicationServices` graph and the default portfolio installs generic operations such as:

```text
artifact.put
claim.validate
evaluate.batch
witness.find
witness.verify
certificate.verify
shrink.run
structure.canonicalize
search.enumerate
experiment.inspect
experiment.wait
experiment.cancel
```

This substrate is the reason Jacobian still needs a parallel `AtomicServiceAdapter` framework, handwritten JSON Schema fragments, `ResultEnvelope` compatibility interpretation, plugin/reference installation machinery, experiment resources, lifecycle routing, and globally destructive `math.run` annotations.

The repair is not to faithfully migrate all of this historical surface into the new typed operation compiler. First decide whether each operation is part of the atomic-math product. Remove the generic workflow substrate from the default runtime and catalog; retain only independently justified mathematical/checker operations. If a research-workflow platform remains useful, move it behind an explicit optional package/profile with its own product contract rather than making every user pay its architectural and runtime cost.

Audit baseline: current `main` at `526575833ef9e849aa5366a55ac5d375d65d36fb`.

## Product contradiction

`docs/explanation/product-blueprint.md` states:

```text
Jacobian is a toolbox of atomic math tools.
Agents find tools and run them; they compose results.
The runtime does not own research strategy.
It is not a workflow engine and not a prescribed explore/verify process.
```

The same document still lists generic workflow primitives such as `artifact.put`, `evaluate.batch`, `witness.find`, `witness.verify`, and `certificate.verify` as illustrative default catalog entries.

Those operations came from a different product model:

```text
claim
  -> plugin evaluator
  -> candidate
  -> witness search
  -> shrink
  -> preservation/checker workflow
  -> durable experiment lifecycle
```

That model may be useful in a benchmark harness or optional research system. It is not a necessary foundation for exact determinants, finite-field arithmetic, polynomial operations, graph invariants, SAT/SMT, Lean checking, or the reusable composition required by #905.

### Every runtime builds the entire application graph

`src/jacobian/runtime/model.py` always executes:

```python
self.core = bootstrap_services(root, options)
self.services = build_application_services(self.core)
self.portfolio = install_portfolio(...)
```

`build_application_services()` constructs all of the following whether or not the installed task needs them:

```text
ClaimValidationService
ClaimDecompositionInstallation
PluginExecutor
StructureService
TransformationService
PolytopeService
EvaluationService
ExperimentService
VerificationService
WitnessSearchService
SearchService
ExperimentRouter
ConjectureService
ShrinkService
ReferenceInstaller
```

The resulting `ApplicationServices` object is effectively a service locator for the former research-workflow product.

### The default portfolio installs the generic services before domain operations

`CoreApplicationInstaller.install()` begins with:

```python
for atomic_adapter in install_atomic_capabilities(ctx, application):
    register_capability(atomic_adapter)
```

It then installs claim decomposition, finite partition/coverage, graph orchestration, domain bundles, exact verification, older polynomial systems, and universal-algebra paths through several independent installers.

Thus the new domain-operation library is not the runtime foundation. It is one installation route beside a large historical application graph.

### `AtomicServiceAdapter` exists to project the workflow services

The generic adapter accepts:

```python
input_schema: dict[str, Any]
output_schema: dict[str, Any]
invoke: Callable[[dict[str, Any]], Any]
```

and reconstructs public execution, scope, completeness, artifacts, and assurance from reflected service-result shapes. Its registrations maintain:

- `_ARTIFACT_URI`, `_CHECKER_URI`, `_EXPERIMENT_URI` schema fragments;
- `_schema()` and `_verification_schema()`;
- manual list→tuple and string→enum conversions;
- result-family-specific verification interpretation;
- explicit artifact callbacks;
- request-field-name-based scope inference.

#1107, #1210, #1212, and #1219 all identify defects in this path. The larger simplification is to question why the path remains a default product substrate at all.

### Experiments contaminate the static math executor

`atomic_experiment_capabilities.py` installs `experiment.cancel` behind the same generic executor used for pure arithmetic. Consequently, `math.run` is conservatively advertised as destructive even for GCD, rank, determinant, or polynomial normalization.

This is not merely missing effect metadata. It is a product-boundary smell: administrative lifecycle controls for a durable research workflow share the primary mathematical execution surface.

### MCP resources expose the same workflow model

The MCP adapter registers:

```text
experiment://{id}
experiment://{id}/accounting
experiment://{id}/scope
experiment://{id}/archive
```

plus workflow-oriented discovery/evidence prompts. A basic math toolbox therefore carries protocol surface, handlers, output models, tests, and lifecycle policy for a durable experiment platform.

## Complexity caused by retaining both products

The atomic-tool product and research-workflow product overlap but have different abstractions:

| Atomic math toolbox | Generic workflow substrate |
| --- | --- |
| typed mathematical requests/results | claim/candidate/plugin URIs |
| agent composes ordinary values | server services coordinate workflow state |
| domain/provider-owned operation | generic evaluator/oracle/reducer capabilities |
| operation-specific bounded execution | durable experiment lifecycle |
| separate checker operation | generic witness/certificate verification envelope |
| reusable exact value contracts | plugin manifest and schema registry conventions |

Keeping both as one default runtime creates parallel answers to the same questions:

```text
How is an operation declared?
How is a request validated?
How is execution represented?
How is a result returned?
How is evidence transported?
How is a checker selected?
How is long-running work controlled?
How is a domain installed?
```

This is why simplifications repeatedly become adapters between two internal products instead of deletion.

## Mature-library comparison

Composable libraries keep the shared protocol smaller than the ecosystem built on it:

- Rust iterators expose one item protocol and adapters; they do not require a universal persisted pipeline runtime for ordinary iteration.
- JAX transformations operate on functions and pytrees; optional higher-level training/orchestration systems remain separate packages.
- scikit-learn exposes estimator/transformer protocols and an explicit `Pipeline`; ordinary estimators do not instantiate a global experiment service graph.
- MCP already provides tools, resources, progress/tasks, and cancellation primitives. A server should expose durable task controls only for operations that actually need them, not as the foundation of all mathematical calls.

The applicable principle is:

> Keep the core around the smallest value/operation protocol. Put optional workflow orchestration outside the core and make its costs explicit.

References:

- https://doc.rust-lang.org/std/iter/
- https://docs.jax.dev/en/latest/jax.tree_util.html
- https://scikit-learn.org/stable/modules/compose.html
- https://modelcontextprotocol.io/specification/draft/server/tools

### 1. Define the default product profile narrowly

A default local Jacobian runtime should contain only:

```text
contract/value registry required by installed operations
provider inventory/readiness
operation compiler/registry
artifact store only where an installed operation needs durable identity
checker registry/trust kernel only when checker packages are authorized
math tools and their resources
```

It should not construct generic claims, plugin evaluation, witness search, shrinking, conjecture, or experiment services merely because they exist in the repository.

### 2. Run a disposition audit over every generic atomic capability

For each current generic ID, choose exactly one:

```text
KEEP_AS_FOUNDATIONAL_MATH_PRIMITIVE
MIGRATE_TO_DOMAIN_OPERATION
KEEP_AS_OPERATOR_INTERNAL
MOVE_TO_OPTIONAL_RESEARCH_PACKAGE
DELETE_NO_CURRENT_PRODUCT_EVIDENCE
```

Required questions:

1. What single mathematical outcome does the operation return?
2. Is it used independently of the old claim/plugin workflow?
3. Does a domain-specific operation already provide the same outcome more clearly?
4. Does the operation expose workflow state rather than mathematics?
5. Is it required by current held-out agent tasks, or only by historical integration tests?
6. Would a fresh user understand and use it without installing a reference plugin/claim artifact first?

Do not migrate an operation into #1212 merely to preserve it.

### 3. Remove workflow services from `ApplicationServices`

After disposition, `build_application_services()` should construct only services required by the selected installed components.

Prefer component-owned installation from #1040 over another large optional-field service locator. A runtime that installs no experiment component has no `ExperimentService`, `ExperimentRouter`, experiment worker, or experiment MCP resource.

### 4. Separate optional research orchestration explicitly

If the generic claim/plugin/search/shrink platform remains valuable, expose it as one of:

```text
optional package/extras group
explicit runtime profile
separate MCP extension/server
benchmark-harness-local subsystem
```

Its tool/resource names, state model, effects, and trust assumptions must be explicit. Enabling it is an operator decision, not default installation.

Do not keep hidden compatibility aliases from the default toolbox under #1214.

### 5. Keep checker infrastructure independent

Deleting the generic workflow must not delete independent mathematical verification.

Domain/checker operations under #1025/#1035 remain usable directly:

```text
producer value/certificate
  -> typed verifier protocol
  -> authorized checker
  -> verdict/record
```

They should not require `ClaimValidationService`, `PluginExecutor`, `WitnessSearchService`, or the reference-plugin workflow unless that exact checker protocol truly needs them.

### 6. Use native MCP task support only where measured

Long operations may expose task/progress/cancellation through their descriptor-owned execution contract (#1077) or direct schema-bearing tool (#1218).

Do not require the custom `search.enumerate` + `experiment.inspect/wait/cancel` protocol for every long-running mathematical operation. Use one explicit durable operation model only where resumability/history is part of the outcome.

### 7. Remove workflow-specific default resources and prompts

A default math toolbox should not advertise experiment URI templates or workflow prompts when no such component is installed.

Generated resource/tool inventory must come from installed components; no central handler should assume the research platform exists.

## Concrete deletion/reduction targets

Subject to the disposition audit, the default path should delete or sharply reduce:

```text
ApplicationServices as a complete workflow service locator
install_atomic_capabilities default registration
AtomicServiceAdapter
atomic_capability_builders.py
atomic_experiment_capabilities.py default installation
claim/evaluation/witness/search/shrink generic schema builders
ExperimentRouter in ordinary runtimes
experiment MCP resource templates in ordinary runtimes
generic workflow prompts
reference plugin installations in the default core
math.run global destructive annotation caused by experiment.cancel
workflow-only tests that force complete runtime hydration
```

Components retained in an optional package may reuse lower-level typed contracts, but must not remain an invisible dependency of the default toolbox.

## Suggested migration

1. Record the installed generic atomic inventory and current invocation/held-out usage.
2. Add a `core_math` runtime fixture that installs domain tools without `build_application_services()`.
3. Prove GCD, matrix rank, polynomial operation, SAT/SMT, Lean, artifact-backed result, and independent checker journeys work in that fixture.
4. Classify every generic atomic capability and publish its disposition.
5. Move/delete claim/plugin/evaluation/witness/shrink/search/experiment components.
6. Make MCP resources/prompts component-contributed rather than globally registered.
7. Change the default runtime/server to the minimal core profile.
8. Remove the old default profile rather than retaining it as a compatibility alias.
9. Re-run startup, context, discovery, first-call validity, and mathematical correctness evaluations.

## Acceptance criteria

- [ ] The default runtime does not construct claim, evaluation, witness-search, generic search, conjecture, shrinking, or experiment services unless an explicitly installed component requires them.
- [ ] Generic workflow capabilities are removed from the default catalog or individually justified as foundational atomic math operations.
- [ ] Pure mathematical calls do not share a globally destructive top-level executor with experiment cancellation.
- [ ] Default MCP resources contain no experiment routes or workflow prompts when the optional workflow component is absent.
- [ ] Independent checker tools work without installing the generic plugin/claim workflow.
- [ ] A basic operation journey does not instantiate reference plugins, durable experiment workers, or unrelated service graphs.
- [ ] The optional workflow product, if retained, has an explicit package/profile/extension identity and is disabled by default.
- [ ] No compatibility shim silently recreates the old complete runtime.
- [ ] Tests use focused component graphs and reserve full workflow hydration for workflow-specific suites.
- [ ] The migration deletes more service/adapter/registration code than it introduces.

## Related issues

- #1148 — product-surface cleanup and host thinness.
- #1207 — delete the managed Skill.
- #1212 — one typed operation protocol; use it only for operations that survive disposition.
- #1214 — no compatibility shims on the pre-stable agent surface.
- #1219 — remove reflective `ResultEnvelope` compatibility.
- #1034 — truthful effects after destructive controls move out of ordinary execution.
- #1040 — component-level installation and lifetime separation.
- #1077 — descriptor-owned execution/task lifecycle.
- #1222 — one-way package dependency graph and optional extension surface.

## Non-goals

- Removing content-addressed artifacts where durable identity, size, replay, or verification genuinely requires them.
- Removing independent checker authorization or verification records.
- Preventing benchmark harnesses from using generic research orchestration.
- Forcing every long-running operation to be synchronous.
- Keeping every historical generic capability merely because tests exist.
- Replacing the removed workflow platform with another generic planner or service bus.

**Comment by morluto** [comment]
## Focused child issues from the deeper trace

- #1229 — delete built-in reference-plugin emulation and the parallel `reference://catalog`; direct domain/checker installation only.
- #1230 — split the concrete local host from optional remote multi-tenant routing/auth/lifecycle.
- #1227 — separate reusable mathematical value identity from backend/run/checker metadata so provider changes do not fork value types.

These should land as deletion-led slices under this product disposition. In particular, #1229 should precede any attempt to migrate reference plugins into the new operation compiler, and #1230 should precede further local-host abstractions that preserve `runtime | tenant_router` dual ownership.

**Comment by morluto** [comment]
## Roadmap integration and implementation gate

#1148 now treats this issue as the first architectural gate after removal of public compatibility surfaces.

The important ordering is:

```text
inventory current generic workflow operations
-> classify each operation
-> delete/move non-product operations
-> only then migrate surviving operations into #1212
```

Do **not** begin by converting the complete `AtomicServiceAdapter` inventory to typed requests/results. That would preserve the former workflow product and make deletion more expensive.

### Direct children / adjacent splits

- #1229 owns deletion of built-in `ReferenceInstaller` / `PluginManifest` emulation and `reference://catalog`.
- #1230 owns separation of the local one-runtime host from optional remote multi-tenancy.
- #1207 owns deletion of the managed Skill.
- #1215 owns removal of adapter recursion inside domain/application code.
- #1212 is the destination only for operations that survive this issue's disposition audit.

### Required inventory output

For every currently installed generic ID, record:

```text
capability ID
single mathematical or lifecycle outcome
current service/adapter owner
runtime services required only because of this operation
current tests/evaluations that use it
proposed disposition
replacement operation/checker ID, if any
production modules and states deleted by the disposition
```

### Migration acceptance addition

A `core_math` fixture is insufficient if it merely bypasses the old services in tests while production still constructs them. The final default `create_runtime()` / local MCP path must prove by instrumentation that a scalar operation, matrix operation, polynomial operation, provider-backed operation, artifact-backed value, and independent checker do not instantiate:

```text
PluginExecutor
ClaimValidationService
EvaluationService
WitnessSearchService
SearchService
ExperimentService / ExperimentRouter
ConjectureService
ShrinkService
ReferenceInstaller
```

unless the selected explicit component actually requires one of them.

**Comment by morluto** [comment]
## Maintainer disposition: make this the first architectural experiment

This issue should precede broad work on #1212, #1218, #1225, and #1223.

The purpose is not to migrate the complete workflow catalog into a cleaner framework. It is to determine which product remains after the old workflow substrate is removed.

### Experiment

Build a focused `core_math` runtime using the smallest currently available direct operation/checker path, then remove the default construction and installation of:

```text
claims and claim decomposition
plugin evaluation
candidate/witness workflows
shrinking and generic search
conjecture and transformation workflow services
experiments and experiment routing
first-party reference-plugin installation
```

Classify every removed generic capability as:

```text
DELETE
MOVE_TO_OPTIONAL_WORKFLOW_PRODUCT
KEEP_AS_OPERATOR_INTERNAL
RETAIN_AS_AN_INDEPENDENT_MATH/CHECKER OPERATION
```

A capability does not survive merely because it has tests or can be migrated into #1212.

### Required before/after evidence

Record:

- default service count and construction graph;
- default installed capability/tool/resource inventory;
- operation-registration mechanisms still in use;
- startup imports, startup time, and optional-provider initialization;
- production files/lines and tests deleted;
- local runtime branches and lifecycle owners removed;
- ordinary math/checker journeys that still pass.

### Success gate

The experiment succeeds only if the default local runtime no longer constructs claim/workflow/plugin/experiment services and the deletion removes substantial machinery such as `AtomicServiceAdapter`, service-locator construction, reference-plugin installation, experiment resources, and workflow-only tests.

Do not add a compatibility profile that silently reconstructs the complete old runtime. Do not require the new operation compiler to exist before proving the deletion.

**Comment by morluto** [comment]
Completed by the workflow-product deletion and #1265 cleanup. The default runtime no longer constructs the generic claim, evaluation, witness-search, conjecture, shrinking, experiment, reference-plugin, or generic finite-partition service graph. Removed IDs/resources are absent, independent checker installation remains direct, and no optional compatibility profile reconstructs the deleted product.
